src/connectors/answers/__tests__/connectAnswers-test.ts

```diff
@@ -0,0 +1,436 @@
+import algoliasearchHelper, { SearchResults } from 'algoliasearch-helper';
+import { createInstantSearch } from '../../../../test/mock/createInstantSearch';
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import {
+  createInitOptions,
+  createRenderOptions,
+} from '../../../../test/mock/createWidget';
+import { createSingleSearchResponse } from '../../../../test/mock/createAPIResponse';
+jest.mock('../../../lib/utils/debounce', () => ({
+  debounce: jest.fn(callback => (...args) => setTimeout(callback(...args), 0)),
+}));
+import { debounce } from '../../../lib/utils/debounce';
+import connectAnswers from '../connectAnswers';
+
+const wait = (time = 10) => {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve();
+    }, time);
+  });
+};
+
+describe('connectAnswers', () => {
+  describe('Usage', () => {
+    it('throws without render function', () => {
+      expect(() => {
+        // @ts-ignore: test connectAnswers with invalid parameters
+        connectAnswers()({});
+      }).toThrowErrorMatchingInlineSnapshot(`
+"The render function is not valid (received type Undefined).
+
+See documentation: https://www.algolia.com/doc/api-reference/widgets/answers/js/#connector"
+`);
+    });
+  });
+
+  const setupTestEnvironment = ({
+    hits,
+    attributesForPrediction = ['description'],
+  }) => {
+    const renderFn = jest.fn();
+    const unmountFn = jest.fn();
+    const client = createSearchClient({
+      // @ts-ignore-next-line
+      initIndex() {
+        return {
+          findAnswers: () => Promise.resolve({ hits }),
+        };
+      },
+    });
+    const instantSearchInstance = createInstantSearch({ client });
+    const makeWidget = connectAnswers(renderFn, unmountFn);
+    const widget = makeWidget({
+      queryLanguages: ['en'],
+      attributesForPrediction,
+    });
+
+    const helper = algoliasearchHelper(client, '', {});
+
+    return {
+      renderFn,
+      unmountFn,
+      instantSearchInstance,
+      widget,
+      helper,
+    };
+  };
+
+  it('is a widget', () => {
+    const render = jest.fn();
+    const unmount = jest.fn();
+
+    const makeWidget = connectAnswers(render, unmount);
+    const widget = makeWidget({
+      queryLanguages: ['en'],
+      attributesForPrediction: ['description'],
```

Comment:
do we have an error message if any of these parameters are missed, or do we rely on the engine erroring?

Reply:
Yeah we throws an error, but the test was missing: https://github.com/algolia/instantsearch.js/pull/4581/commits/92e9a996f7f1ba4fda1b73070f2963952edf0bbf